lib/irb/debug/ui.rb

```diff
@@ -0,0 +1,106 @@
+require 'io/console/size'
+require 'debug/console'
+
+module IRB
+  module Debug
+    class UI < DEBUGGER__::UI_Base
+      PROMPT_NAME = "irb:rdbg"
+
+      def initialize(thread, scanner)
+        @thread = thread
+        @scanner = scanner
+      end
+
+      def remote?
+        false
+      end
+
+      def activate session, on_fork: false
+      end
+
+      def deactivate
+      end
+
+      def width
+        if (w = IO.console_size[1]) == 0 # for tests PTY
+          80
+        else
+          w
+        end
+      end
+
+      def quit n
+        yield
+        exit n
+      end
+
+      def ask prompt
+        setup_interrupt do
+          print prompt
+          ($stdin.gets || '').strip
+        end
+      end
+
+      def puts str = nil
+        case str
+        when Array
+          str.each{|line|
+            $stdout.puts line.chomp
+          }
+        when String
+          str.each_line{|line|
+            $stdout.puts line.chomp
+          }
+        when nil
+          $stdout.puts
+        end
+      end
+
+      def readline _
+        setup_interrupt do
+          tc = DEBUGGER__::SESSION.get_thread_client(@thread)
+          cmd = IRB.debug_readline(binding: tc.current_frame.binding || TOPLEVEL_BINDING, scanner: @scanner, prompt_name: PROMPT_NAME)
+
+          case cmd
+          when nil # when user types C-d
+            "exit!"
```

Comment:
In normal irb, `C-d` and `exit` will exit irb session, not the program.
Maybe `continue` instead of `exit!`?
Also, is it possible to quit debug mode and go back to normal irb mode with `continue`?

Reply:
> Maybe continue instead of exit!?

Good point! I'll change it.

> is it possible to quit debug mode and go back to normal irb mode with continue?

Can you describe a use case for that? Since `irb:rdbg` is basically extending IRB session with debugging functionality, why do we want to roll it back?

Reply:
> use case

Sorry, I don't have it. I misunderstood that `continue` ends debugging and want irb to go back to normal mode. actually it continues the execution and stops at the next breakpoint.